### DISCUSSION TITLE
how to connect to db only once

### DISCUSSION THREAD

**Discussion by MutedOne** [sidorares/node-mysql2#2215] [General]
im new to this. how to connect to db in mysql2 once? i see everytime the file call it create new connection
![image](https://github.com/sidorares/node-mysql2/assets/47779436/7bf1fed2-ba01-49be-994d-6aecc73e0205)

SHOW STATUS LIKE 'Threads_connected';

any discord server that i can join?

**Comment by sidorares** [comment]
Can you show your `setup.js`? I assume it creates a pool. By default the pool automatically creates [up to 10 connections](https://github.com/sidorares/node-mysql2/blob/2595f9f0869d43531d493764ea7d89a868a7597e/lib/pool_config.js#L16) but you can customise that by passing `connectionsLimit` when you create a pool. If you set it to 1 your pool would have at most 1 open connection
